### ISSUE TITLE
[BUG] 1.4.0 topological_sort(g) is broken.

### ISSUE THREAD

**Issue by sbromberger** [JuliaGraphs/LightGraphs.jl#1337]
The new `topological_sort` is broken in master.
```
julia> g                                                                                                                                     
{3, 2} directed simple Int64 graph                                                                                                           
julia> collect(edges(g))
2-element Array{LightGraphs.SimpleGraphs.SimpleEdge{Int64},1}:
 Edge 1 => 2
 Edge 3 => 1
julia> LightGraphs.Traversals.topological_sort(g)
5-element Array{Int64,1}:
 3
 1
 2
 1
 2
```

**Comment by alimamdouh212** [comment]
the problem is that the search starts from arbitrary nodes, it should start from the which has no inneighbours

**Comment by alimamdouh212** [comment]
@sbromberger i think i have solved the bug in #1340, it produce the correct result for that case, would  you please review it.

**Comment by sbromberger** [comment]
Closed via #1340
